File: Yow.CoD.Finance.Domain/Contracts/TransactionReceipt.cs
using System;

namespace Yow.CoD.Finance.Domain.Contracts
{
    public sealed class TransactionReceipt : Receipt
    {
        public TransactionReceipt(
            Guid aggregateId, 
            int version,
            string transactionId) 
            : base(aggregateId, version)
        {
            TransactionId = transactionId;
        }

        public string TransactionId { get; }
    }
}

File: typescript/packages/domain/src/contracts/transaction-receipt.ts
import { Receipt } from "./receipt.js";

export class TransactionReceipt extends Receipt {
  readonly transactionId: string;

  constructor(aggregateId: string, version: number, transactionId: string) {
    super(aggregateId, version);
    this.transactionId = transactionId;
  }
}
